File: apps/dsa-web/src/api/ocr.ts
import apiClient from './index';
import { toCamelCase } from './utils';

export interface OcrRecognizeResponse {
  success: boolean;
  codes: string[];
  count: number;
  provider: string | null;
  error: string | null;
}

export const ocrApi = {
  /**
   * 识别股票截图中的代码
   * @param imageBase64 Base64 编码的图片数据（不含 data:image/xxx;base64, 前缀）
   * @param mimeType 图片 MIME 类型
   * @returns 识别结果
   */
  recognize: async (imageBase64: string, mimeType: string = 'image/png'): Promise<OcrRecognizeResponse> => {
    const response = await apiClient.post<Record<string, unknown>>(
      '/api/v1/ocr/recognize',
      {
        image_base64: imageBase64,
        mime_type: mimeType,
      }
    );
    return toCamelCase<OcrRecognizeResponse>(response.data);
  },
};

File: api/v1/schemas/ocr.py
# -*- coding: utf-8 -*-
"""
===================================
OCR 识别相关 Schema
===================================

定义 OCR 接口的请求和响应模型
"""

from typing import List, Optional
from pydantic import BaseModel, Field


class OcrRequest(BaseModel):
    """OCR 识别请求"""
    image_base64: str = Field(..., description="Base64 编码的图片数据（不含 data:image/xxx;base64, 前缀）")
    mime_type: str = Field(default="image/png", description="图片 MIME 类型")


class OcrResponse(BaseModel):
    """OCR 识别响应"""
    success: bool = Field(..., description="是否识别成功")
    codes: List[str] = Field(default_factory=list, description="识别到的股票代码列表")
    count: int = Field(default=0, description="识别到的代码数量")
    provider: Optional[str] = Field(None, description="使用的 OCR 提供商（gemini/anthropic）")
    error: Optional[str] = Field(None, description="错误信息")
